File: frontend-web/src/app/core/services/reservas.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';
import { Reserva, ReservaCreate } from '../models/reserva.model';

@Injectable({ providedIn: 'root' })
export class ReservasService {
  private http = inject(HttpClient);
  private readonly base = `${environment.apiUrl}/reservas`;

  crear(data: ReservaCreate) {
    return this.http.post<Reserva>(this.base, data);
  }

  misReservas() {
    return this.http.get<Reserva[]>(`${this.base}/mis-reservas`);
  }

  obtener(id: number) {
    return this.http.get<Reserva>(`${this.base}/${id}`);
  }

  cancelar(id: number) {
    return this.http.patch<Reserva>(`${this.base}/${id}/cancelar`, {});
  }

  // ===== MÉTODOS PARA ENCARGADO =====

  miSucursal() {
    return this.http.get<Reserva[]>(`${this.base}/mi-sucursal`);
  }

  detalleEncargado(reservaId: number) {
    return this.http.get<Reserva>(`${this.base}/detalle/${reservaId}`);
  }

  cambiarEstado(reservaId: number, nuevoEstado: string) {
    return this.http.patch<Reserva>(
      `${this.base}/${reservaId}/estado?nuevo_estado=${nuevoEstado}`,
      {}
    );
  }
}

File: mobile-app/lib/core/services/reservas_service.dart
import '../models/reserva.dart';
import '../network/dio_client.dart';

class ReservasService {
  final DioClient _dioClient = DioClient();

  Future<Reserva> crear(ReservaCreate data) async {
    final response = await _dioClient.post(
      '/reservas',
      data: data.toJson(),
    );
    return Reserva.fromJson(response.data as Map<String, dynamic>);
  }

  Future<List<Reserva>> misReservas() async {
    final response = await _dioClient.get('/reservas/mis-reservas');
    if (response.data is List) {
      return (response.data as List)
          .map((json) => Reserva.fromJson(json as Map<String, dynamic>))
          .toList();
    }
    return [];
  }

  Future<Reserva> obtener(int id) async {
    final response = await _dioClient.get('/reservas/$id');
    return Reserva.fromJson(response.data as Map<String, dynamic>);
  }

  Future<Reserva> cancelar(int id) async {
    final response = await _dioClient.patch('/reservas/$id/cancelar');
    return Reserva.fromJson(response.data as Map<String, dynamic>);
  }
}
